spec/models/manageiq/providers/openstack/cloud_manager/provision/volume_attachment_spec.rb

```diff
@@ -1,82 +1,102 @@
 describe ManageIQ::Providers::Openstack::CloudManager::Provision::VolumeAttachment do
-  before do
-    @ems = FactoryBot.create(:ems_openstack_with_authentication)
-    @template = FactoryBot.create(:template_openstack, :ext_management_system => @ems)
-    @flavor = FactoryBot.create(:flavor_openstack)
-    @volume = FactoryBot.create(:cloud_volume_openstack)
+  let(:ems)          { FactoryBot.create(:ems_openstack_with_authentication) }
+  let(:flavor)       { FactoryBot.create(:flavor_openstack) }
+  let(:template)     { FactoryBot.create(:template_openstack, :ext_management_system => ems) }
+  let(:volume)       { FactoryBot.create(:cloud_volume_openstack) }
+  let(:service)      { double("volume service") }
+  let(:task_options) { {:instance_type => flavor, :src_vm_id => template.id, :volumes => task_volumes} }
+  let(:task_volumes) { [{:name => "custom_volume_1", :size => 2}] }
+  let(:task)         { FactoryBot.create(:miq_provision_openstack, :source => template, :state => 'pending', :status => 'Ok', :options => task_options) }
 
+  before do
     # We're storing objects in the instance_type, so we must permit loading this class
-    ActiveRecord.yaml_column_permitted_classes = YamlPermittedClasses.app_yaml_permitted_classes | [@flavor.class]
-
-    @task = FactoryBot.create(:miq_provision_openstack,
-                               :source  => @template,
-                               :state   => 'pending',
-                               :status  => 'Ok',
-                               :options => {
-                                 :instance_type => @flavor,
-                                 :src_vm_id     => @template.id,
-                                 :volumes       => [{:name => "custom_volume_1", :size => 2}]
-                               })
+    ActiveRecord.yaml_column_permitted_classes = YamlPermittedClasses.app_yaml_permitted_classes | [flavor.class]
   end
 
   context "#configure_volumes" do
-    it "create volumes" do
-      service = double
-      allow(service).to receive_message_chain('volumes.create').and_return @volume
-      allow(@task.source.ext_management_system).to receive(:with_provider_connection)\
+    before do
+      allow(service).to receive_message_chain('volumes.create').and_return(volume)
```

Comment:
I see this was here already, but I generally prefer

```suggestion
      allow(service).to receive_message_chain(:volumes, :create).and_return(volume)
```

Reply:
Yeah it was here previously, I do have a "cleanup specs" commit in here already though so I'd rather change all of them to be consistent